open_seq2seq/decoders/rnn_decoders.py

```diff
@@ -172,16 +176,29 @@ def _decode(self, input_dict):
     tgt_lengths = input_dict['target_tensors'][1] if 'target_tensors' in \
                                                      input_dict else None
 
-    self._dec_emb_w = tf.get_variable(
-        name='DecoderEmbeddingMatrix',
-        shape=[self._tgt_vocab_size, self._tgt_emb_size],
-        dtype=tf.float32,
-    )
+    # self._dec_emb_w = tf.get_variable(
+    #     name='DecoderEmbeddingMatrix',
+    #     shape=[self._tgt_vocab_size, self._tgt_emb_size],
+    #     dtype=tf.float32,
+    # )
 
     self._output_projection_layer = tf.layers.Dense(
         self._tgt_vocab_size, use_bias=False,
     )
 
+    if not self._weight_tied:
+      self._dec_emb_w = tf.get_variable(
+          name='DecoderEmbeddingMatrix',
+          shape=[self._tgt_vocab_size, self._tgt_emb_size],
+          dtype=tf.float32
+      )
+    else:
+      fake_input = tf.zeros(shape=(1, self._tgt_emb_size))
+      fake_output = self._output_projection_layer.apply(fake_input)
+      with tf.variable_scope("dense", reuse=True):
```

Comment:
see if we can use get_weights

Reply:
@chiphuyen

Reply:
the method is only available in TF 1.9+. It doesn't work if we're currently on TF 1.8